File: Documentation/4.0/Samples/nodejs/indexes/sorting.js
import * as assert from "assert";
import { 
    DocumentStore,
    PutIndexesOperation,
    AbstractIndexCreationTask,
    IndexDefinition
} from "ravendb";

const store = new DocumentStore();
const session = store.openSession();

{
    //region static_sorting2
    class Products_ByName extends AbstractIndexCreationTask {
        constructor() {
            super();

            this.map = `docs.Products.Select(product => new {     
                Name = product.Name 
            })`;

            this.analyze("Name", 
                "Raven.Server.Documents.Indexes.Persistence.Lucene.Analyzers.Collation.Cultures.SvCollationAnalyzer, Raven.Server");
        }
    }
    //endregion
}

File: Documentation/4.1/Samples/java/src/test/java/net/ravendb/Indexes/Sorting.java
package net.ravendb.Indexes;

import net.ravendb.client.documents.indexes.AbstractIndexCreationTask;

public class Sorting {
    //region static_sorting2
    public static class Products_ByName extends AbstractIndexCreationTask {
        public Products_ByName() {
            map = "docs.Products.Select(product => new { " +
                "    Name = product.Name " +
                "})";

            analyze("Name", "Raven.Server.Documents.Indexes.Persistence.Lucene.Analyzers.Collation.Cultures.SvCollationAnalyzer, Raven.Server");
        }
    }
    //endregion

}
